### PR TITLE
Chap 11 - Account activation

### PR DESCRIPTION
**Pull request by namkidd**

### CODE REVIEW

**Comment by namkidd** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -28,15 +31,40 @@ def new_token
...
+    BCrypt::Password.new(digest).is_password? token
   end
 
   def forget
     update_attributes remember_digest: nil
   end
+
+  def activate
+    update_attributes activated: true
+    update_attributes activated_at: Time.zone.now
```
Comment (huongnguyenmta) [inline app/models/user.rb]: gộp dòng 52, 53

File: app/views/user_mailer/account_activation.html.erb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<h1><%= t ".sample" %></h1>
+<p> <%= t ".hi" %><%= @user.name %>,</p>
+<p><%= t ".hello" %></p>
+<%= link_to t(".activate"),
+ (edit_account_activation_url @user.activation_token, email: @user.email) %>
```
Comment (huongnguyenmta) [inline app/views/user_mailer/account_activation.html.erb]: `edit_account_activation_url(@user.activation_token, email: @user.email)`

File: app/views/user_mailer/password_reset.html.erb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<h1>User#password_reset</h1>
```
Comment (huongnguyenmta) [inline app/views/user_mailer/password_reset.html.erb]: I18n

File: app/views/user_mailer/password_reset.html.erb [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+
+<p>
+  <%= @greeting %>, find me in app/views/user_mailer/password_reset.html.erb
```
Comment (huongnguyenmta) [inline app/views/user_mailer/password_reset.html.erb]: I18n

File: app/views/user_mailer/password_reset.text.erb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+User#password_reset
```
Comment (huongnguyenmta) [inline app/views/user_mailer/password_reset.text.erb]: I18n

File: config/environments/production.rb [hunk 6]
```diff
@@ -14,6 +14,19 @@
   config.i18n.fallbacks = true
   config.active_support.deprecation = :notify
   config.log_formatter = ::Logger::Formatter.new
+  config.action_mailer.raise_delivery_errors = true
+  config.action_mailer.delivery_method = :smtp
+  host = '<your heroku app>.herokuapp.com'
+  config.action_mailer.default_url_options = { host: host }
+  ActionMailer::Base.smtp_settings = {
+    :address        => 'smtp.sendgrid.net',
```
Comment (huongnguyenmta) [inline config/environments/production.rb]: `address: "smtp.sendgrid.net"`

**Comment by huongnguyenmta** [comment]
commented

**Comment by namkidd** [comment]
ready

**Comment by hungnh103** [comment]
commented
Em merge nhé

**Pull request merged**